synapse/lib/jsonstor.py

```diff
@@ -247,8 +262,56 @@ async def cmpDelPathObjProp(self, path, prop, valu):
         self.dirty[buid] = item
         return True
 
+    async def popPathObjProp(self, path, prop, defv=None):
+
+        buid = self._pathToBuid(path)
+        if buid is None:
+            return defv
+
+        item = self._getBuidItem(buid)
+        if item is None:
+            return defv
+
+        step = item
+        names = self._pathToTupl(prop)
+        for name in names[:-1]:
+            step = step[name]
```

Comment:
Does this need to be
`           step = step.get(name, s_common.novalu)
            if step is s_common.novalu:
                return None`